### PR TITLE
Fixes CheckBoxField title being read twice on Voice Over 

### PR DESCRIPTION
**Pull request by gt106551**
Summary
Terra - Form Checkbox Field - VO announces group and legend multiple times

Solution
I have replaced legend Tag with div  Tag in Safari & Edge.

Testing
I have tested with Chrome, Safari,Edge.

### CODE REVIEW

File: packages/terra-form-checkbox/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,11 @@
 
 ## Unreleased
 
+## 4.21.0 - (May 18, 2023)
+
+* Fixed
+  * Fixed issue - Form Checkbox Field - VO announces group and legend multiple times.
```
Comment (sdadn) [inline packages/terra-form-checkbox/CHANGELOG.md:8]:
```suggestion
  * Fixed issue with VO announcing group and legend multiple times.
```

**Review by sdadn — CHANGES_REQUESTED** [review]

File: packages/terra-form-checkbox/CHANGELOG.md (hunk 1 above)
Comment (sdadn) [inline packages/terra-form-checkbox/CHANGELOG.md:8]:
```suggestion
  * Fixed issue with VO announcing group and legend multiple times.
```

File: packages/terra-form-checkbox/CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,11 @@
 
 ## Unreleased
 
+## 4.21.0 - (May 18, 2023)
```
Comment (sdadn) [inline packages/terra-form-checkbox/CHANGELOG.md:5]: This gets added automatically when releasing so we don't need to add this right now.
```suggestion
```

**Review by sdadn — CHANGES_REQUESTED** [review]

File: packages/terra-form-checkbox/src/CheckboxField.jsx [hunk 3]
```diff
@@ -114,24 +114,38 @@ const CheckboxField = (props) => {
...
+    </div>
+  );
+
+  const legendGroup = isSafariBrowser ? (
+    <div id={legendAriaDescriptionId} className={cx(['legend-group', { 'legend-group-hidden': isLegendHidden }])}>
+      {legendCode}
+    </div>
+  ) : (
+    <legend id={legendAriaDescriptionId} className={cx(['legend-group', { 'legend-group-hidden': isLegendHidden }])}>
+      {legendCode}
```
Comment (sdadn) [inline packages/terra-form-checkbox/src/CheckboxField.jsx:148]: Can you add some jest unit tests for when isSafariBrowser is true?

**Review by sdadn — CHANGES_REQUESTED** [review]

**Comment by supreethmr** [comment]
can you simply the if else block written for div and legend like below

```jsx
  const isSafariOREdge = ((navigator.userAgent.indexOf('Safari') !== -1 || navigator.userAgent.indexOf('Edg') !== -1) && navigator.userAgent.indexOf('Chrome') === -1);

  const Component = (isSafariOREdge) ? 'div' : 'legend';
  const legendGroup = (
    <Component id={legendAriaDescriptionId} className={cx(['legend-group', { 'legend-group-hidden': isLegendHidden }])}>
      <div {...legendAttrs} className={legendClassNames}>
        {isInvalid && <span className={cx('error-icon')} />}
        {required && (isInvalid || !hideRequired) && (
          <React.Fragment>
            <div aria-hidden="true" className={cx('required')}>*</div>
            <VisualyHiddenText text={intl.formatMessage({ id: 'Terra.form.field.required' })} />
          </React.Fragment>
        )}
        {legend}
        {required && !isInvalid && hideRequired && <span className={cx('required-hidden')}>*</span>}
        {showOptional && !required
          && (
            <span className={cx('optional')}>{intl.formatMessage({ id: 'Terra.form.field.optional' })}</span>
          )}
        {!isInvalid && <span className={cx('error-icon-hidden')} />}
      </div>
    </Component>
  );
```

File: packages/terra-form-checkbox/src/CheckboxField.jsx [hunk 4]
```diff
@@ -114,8 +114,14 @@ const CheckboxField = (props) => {
   const errorAriaDescriptionId = error ? `terra-checkbox-field-description-error-${uniqueid()}` : '';
   const ariaDescriptionIds = `${legendAriaDescriptionId} ${errorAriaDescriptionId} ${helpAriaDescriptionId}`;
 
+  let isSafariOrEdgBrowser = false;
+
+  if ((navigator.userAgent.indexOf('Safari') !== -1 && navigator.userAgent.indexOf('Chrome') === -1) || navigator.userAgent.indexOf('Edg') !== -1) {
+    isSafariOrEdgBrowser = true;
+  }
```
Comment (supreethmr) [inline packages/terra-form-checkbox/src/CheckboxField.jsx:121]:
```suggestion
  const isSafariOREdge = ((navigator.userAgent.indexOf('Safari') !== -1 && navigator.userAgent.indexOf('Chrome') === -1) || navigator.userAgent.indexOf('Edg') !== -1);
```

File: packages/terra-form-checkbox/src/CheckboxField.jsx [hunk 5]
```diff
@@ -114,8 +114,14 @@ const CheckboxField = (props) => {
...
+  const Component = (isSafariOrEdgBrowser) ? 'div' : 'legend';
```
Comment (supreethmr) [inline packages/terra-form-checkbox/src/CheckboxField.jsx:122]:
```suggestion
  const Component = (isSafariOREdge) ? 'div' : 'legend';
```

File: packages/terra-form-checkbox/tests/jest/CheckboxField.test.jsx [hunk 6]
```diff
@@ -8,12 +8,23 @@ import Checkbox from '../../src/Checkbox';
...
+  userAgentGetter = jest.spyOn(window.navigator, 'userAgent', 'get');
+});
+
 it('should render a default checkbox field', () => {
   const checkBox = (<CheckboxField legend="Default CheckboxField" />);
   const wrapper = shallowWithIntl(checkBox);
   expect(wrapper).toMatchSnapshot();
 });
 
+it('should render checkbox field with div element for safari browser', () => {
```
Comment (supreethmr) [inline packages/terra-form-checkbox/tests/jest/CheckboxField.test.jsx:22]: Can we have add similar tests for chrome and edge as well

File: packages/terra-form-checkbox/tests/jest/CheckboxField.test.jsx [hunk 7]
```diff
@@ -8,12 +8,23 @@ import Checkbox from '../../src/Checkbox';
...
+  userAgentGetter.mockReturnValue('Safari');
+  const wrapper = shallowWithIntl(<checkboxField legend="Coustom Message CheckboxField" />);
+  expect(wrapper).toMatchSnapshot();
```
Comment (supreethmr) [inline packages/terra-form-checkbox/tests/jest/CheckboxField.test.jsx:25]:
```suggestion
  expect(wrapper.dive()).toMatchSnapshot();
```

File: packages/terra-form-checkbox/src/CheckboxField.jsx (hunk 3 above)
Comment (gt106551) [reply packages/terra-form-checkbox/src/CheckboxField.jsx]: i have added test cases.

File: packages/terra-form-checkbox/tests/jest/__snapshots__/CheckboxField.test.jsx.snap [hunk 8]
```diff
@@ -567,6 +567,72 @@ exports[`should render an optional part on the label 1`] = `
...
+      "now": [Function],
+      "onError": [Function],
+      "textComponent": "span",
+      "timeZone": null,
+    }
+  }
+  legend="Coustom Message CheckboxField"
+/>
+`;
+
```
Comment (sdadn) [inline packages/terra-form-checkbox/tests/jest/__snapshots__/CheckboxField.test.jsx.snap:602]: These snapshots look incorrect; you'll need to use a `.dive()` statement to get the inner contents of the rendered components.

File: packages/terra-form-checkbox/tests/jest/CheckboxField.test.jsx [hunk 9]
```diff
@@ -8,12 +8,30 @@ import Checkbox from '../../src/Checkbox';
...
+
 it('should render a default checkbox field', () => {
   const checkBox = (<CheckboxField legend="Default CheckboxField" />);
   const wrapper = shallowWithIntl(checkBox);
   expect(wrapper).toMatchSnapshot();
 });
 
+it('should render checkbox field with div element for safari browser or Edg browser', () => {
+  userAgentGetter.mockReturnValue('safari Edg');
+  const wrapper = shallowWithIntl(<checkboxField legend="Coustom Message CheckboxField" />);
```
Comment (sdadn) [inline packages/terra-form-checkbox/tests/jest/CheckboxField.test.jsx:24]: To get the inner contents of the component:
```suggestion
  const wrapper = shallowWithIntl(<checkboxField legend="Coustom Message CheckboxField" />).dive();
```

File: packages/terra-form-checkbox/tests/jest/CheckboxField.test.jsx [hunk 10]
```diff
@@ -8,12 +8,30 @@ import Checkbox from '../../src/Checkbox';
...
+  expect(wrapper).toMatchSnapshot();
```
Comment (sdadn) [inline packages/terra-form-checkbox/tests/jest/CheckboxField.test.jsx:25]: We shouldn't rely purely on snapshots for testing as they can be overwritten and incorrect snapshots won't register as failures. There should be an additional expect statement to explicitly test for the div within the wrapper.

File: packages/terra-form-checkbox/tests/jest/CheckboxField.test.jsx [hunk 11]
```diff
@@ -8,12 +8,30 @@ import Checkbox from '../../src/Checkbox';
...
+});
+
+it('should render checkbox field with legend element for Chrome browser', () => {
+  userAgentGetter.mockReturnValue('Chrome');
+  const wrapper = shallowWithIntl(<checkboxField legend="Coustom Message CheckboxField" />);
+  expect(wrapper).toMatchSnapshot();
```
Comment (sdadn) [inline packages/terra-form-checkbox/tests/jest/CheckboxField.test.jsx:31]: Same comments as for the other test.

File: packages/terra-form-checkbox/src/CheckboxField.jsx (hunk 4 above)
Comment (gt106551) [reply packages/terra-form-checkbox/src/CheckboxField.jsx]: i have made changes as per suggestions.

File: packages/terra-form-checkbox/tests/jest/CheckboxField.test.jsx (hunk 6 above)
Comment (gt106551) [reply packages/terra-form-checkbox/tests/jest/CheckboxField.test.jsx]: oky

File: packages/terra-form-checkbox/tests/jest/CheckboxField.test.jsx (hunk 7 above)
Comment (gt106551) [reply packages/terra-form-checkbox/tests/jest/CheckboxField.test.jsx]: i have made changes as per suggestions.

File: packages/terra-form-checkbox/CHANGELOG.md [hunk 12]
```diff
@@ -2,6 +2,9 @@
 
 ## Unreleased
 
+* Fixed
+  * Fixed issue with VO announcing group and legend multiple times.
```
[the remark this answers is not in the document]
Comment (gt106551) [reply packages/terra-form-checkbox/CHANGELOG.md]: i have made changes as per suggestions.

File: packages/terra-form-checkbox/tests/jest/__snapshots__/CheckboxField.test.jsx.snap (hunk 8 above)
Comment (gt106551) [reply packages/terra-form-checkbox/tests/jest/__snapshots__/CheckboxField.test.jsx.snap]: i have made changes as per suggestions.

File: packages/terra-form-checkbox/tests/jest/CheckboxField.test.jsx (hunk 10 above)
Comment (gt106551) [reply packages/terra-form-checkbox/tests/jest/CheckboxField.test.jsx]: i have made changes as per suggestions.

**Review by supreethmr — APPROVED** [review]

**Review by supreethmr — APPROVED** [review]

**Review by sugan2416 — APPROVED** [review]
